src/alchemlyb/estimators/bar_.py

```diff
@@ -0,0 +1,135 @@
+            configuration n evaluated at state k.
+
+        """
+        # sort by state so that rows from same state are in contiguous blocks
+        u_nk = u_nk.sort_index(level=u_nk.index.names[1:])
+
+        # get a list of the lambda states
+        self.states_ = u_nk.columns.values.tolist()
+
+        # group u_nk by lambda states
+        groups = u_nk.groupby(level=u_nk.index.names[1:])
+        N_k = [(len(groups.get_group(i)) if i in groups.groups else 0) for i in u_nk.columns]
+
+        # Now get free energy differences and their uncertainties for each step
+        deltas = np.array([])
+        d_deltas = np.array([])
+        for k in range(len(N_k) - 1):
+            # get us from lambda step k
+            uk = groups.get_group(self.states_[k])
+            # get w_F
+            w_f = uk.iloc[:, k+1] - uk.iloc[:, k]
+
+            # get us from lambda step k+1
+            uk1 = groups.get_group(self.states_[k+1])
+            # get w_R
+            w_r = uk1.iloc[:, k] - uk1.iloc[:, k+1]
+
+            # now determine df and ddf using pymbar.BAR
+            df, ddf = BAR_(w_f, w_r,
+                             method=self.method,
+                             maximum_iterations=self.maximum_iterations,
+                             relative_tolerance=self.relative_tolerance,
+                             verbose=self.verbose)
+
+            deltas = np.append(deltas, df)
+            d_deltas = np.append(d_deltas, ddf**2)
+
+        # build matrix of deltas between each state
+        adelta = np.zeros((len(deltas) + 1, len(deltas) + 1))
+        ad_delta = np.zeros_like(adelta)
+
+        for j in range(len(deltas)):
+            out = []
+            dout = []
+            for i in range(len(deltas) - j):
+                out.append(deltas[i:i + j + 1].sum())
+
+                # See https://github.com/alchemistry/alchemlyb/pull/60#issuecomment-430720742
+                # Error estimate generated by BAR ARE correlated
+
+                # Use the BAR uncertainties between two neighbour states
+                if j == 0:
+                    dout.append(d_deltas[i:i + j + 1].sum())
+                # Other uncertainties are unknown at this point
+                else:
+                    dout.append(float('NaN'))
+
+            adelta += np.diagflat(np.array(out), k=j + 1)
+            ad_delta += np.diagflat(np.array(dout), k=j + 1)
+
+        # yield standard delta_f_ free energies between each state
+        self.delta_f_ = pd.DataFrame(adelta - adelta.T,
+                                     columns=self.states_,
+                                     index=self.states_)
+
+        # yield standard deviation d_delta_f_ between each state
+        self.d_delta_f_ = pd.DataFrame(np.sqrt(ad_delta + ad_delta.T),
+                                       columns=self.states_,
+                                       index=self.states_)
+
+        return self
+
+    def predict(self, u_ln):
```

Comment:
Can we drop it?

(It's neither documented nor tested.)

Reply:
If you keep then there should be a test that shows that it does nothing.

Reply:
This is comes from the implementation of MBAR which I used as a starting point. I don't find any usages of this method in the project so let's drop it...

Reply:
If it's in the MBAR class then we should also remove it from there (in a different PR).